zfsbootmenu/lib/zfsbootmenu-core.sh

```diff
@@ -1673,6 +1673,16 @@ load_key() {
     return 0
   fi
 
+  # Run early load_key hooks, if they exist
+  if [ -d /libexec/load_key.d ]; then
+    tput clear
+    for _hook in /libexec/load_key.d/*; do
+      zinfo "Processing hook: ${_hook}"
+      [ -x "${_hook}" ] && "${_hook}" "${encroot}"
```

Comment:
```suggestion
      [ -x "${_hook}" ] || continue
      if ! "${_hook}" "${encroot}"; then
        zwarn "load-key hook failed for '${encroot}', aborting load attempt"
        return 1
      fi
```
Let's use the return code of hooks to signal a failure condition that aborts key loading altogether.

Reply:
Agree